src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -335,7 +335,7 @@ void AbstractWriteLog<I>::arm_periodic_stats() {
 }
 
 template <typename I>
-void AbstractWriteLog<I>::update_entries(std::shared_ptr<GenericLogEntry> log_entry,
+void AbstractWriteLog<I>::update_entries(std::shared_ptr<GenericLogEntry> &log_entry,
```

Comment:
Nit: we try to use pointers for in-out params in librbd:
```suggestion
void AbstractWriteLog<I>::update_entries(std::shared_ptr<GenericLogEntry>* log_entry,
```

Reply:
@dillaman 
We can use the pointer instead of the reference here. But, it seems a little strange that we use a pointer to point to the shared_ptr object.

Reply:
Why? It's basically the same as a `**` type (pointer to pointer) which we use in librbd for in-out.

Reply:
OK. I need to add more changes here when using pointer here.

Reply:
@dillaman 
Done. BTW, doesn't it seem strange that we use pointer type(for in-out, e.g. log_entry) and reference type(for in-out, e.g. sync_point_entries) in the same API?

Reply:
Yes, it is strange. Care to clean that up as well?

Reply:
@dillaman 
finish rebasing to solve the conflict.
Note: Some other APIs for PWL feature also doesn't follow using pointer type. It's better to file another PR to solve this code style problem.